pb/sensorsinfo_proto/sensorsinfo.proto

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Protocol buffers for storing active connection information.
+syntax = "proto3";
+
+package sensorsinfo;
+
+option go_package = "github.com/googleinterns/sensor-historian/pb/sensorsinfo_proto;sensorsinfo";
+
+// ActiveConn contains information about an active connection.
+message ActiveConn{
+    int32 Number = 1;
+    string OperatingMode = 2;
+    string PackageName = 3;
+    int32 UID = 4;
+    int32 SensorNumber = 5;
+    int32 PendingFlush = 6;
+    double SamplingRateHz = 7;
+    double BatchingPeriodS = 8;
+    bool HasSensorserviceRecord = 9;
+    string Source = 10;
+}
+
+// SubscriptionInfo contains information about one subscription event of
+// a sensor to an application.
+message SubscriptionInfo{
+    int64 StartMs = 1;
+    int64 EndMs = 2;
+    int32 SensorNumber = 3;
+    int32 UID = 4;
+    string PackageName = 5;
+    double SamplingRateHz = 6;
+    double BatchingPeriodS = 7;
+    string Source = 8;
+}
+
+enum RequestMode{
+    CONTINUOUS = 0;
+    ON_CHANGE = 1;
+    ONE_SHOT = 2;
+    SPECIAL = 3;
+}
+
+message Sensor{
+    string Name = 1;
+    string Type = 2;
+    int32 Version = 3;
```

Comment:
Are all of these fields required?

Reply:
Can we mark some of these as optional

Reply:
So in this protobuf file,  I am using the syntax for "proto3", which removes the concepts of "required" and "optional". In this case all the fields are marked automatically as "optional“。
Since all the old protobuf files defined for battery historian are all using "proto2" and all fields are marked "optional", I will change my data structures accordingly and have all of them as "optional" too.

Reply:
After changing the file to proto2 syntax, the go file generated is very different from what I have before. Specifically, even if I set all the fields to be type string, int, bool, etc. the go file uses pointers instead. It is inconvenient to have everything in pointers, so shall we stick with the "proto3" syntax and do not set things to optional?
In the current implementation, all the fields are optional so it is still flexible for other layers.

Reply:
Can you update it with proto3. Latest pull request has proto2 format.